src/main/java/org/jabref/gui/preferences/importexport/ImportExportTabViewModel.java

```diff
@@ -132,4 +157,67 @@ public BooleanProperty grobidEnabledProperty() {
     public StringProperty grobidURLProperty() {
         return grobidURLProperty;
     }
+
+    public ListProperty<FetcherApiKey> fetcherApiKeys() {
+        return apiKeys;
+    }
+
+    public ObjectProperty<FetcherApiKey> selectedApiKeyProperty() {
+        return selectedApiKeyProperty;
+    }
+
+    public void checkCustomApiKey() {
+        final String apiKeyName = selectedApiKeyProperty.get().getName();
+
+        final Optional<CustomizableKeyFetcher> fetcherOpt =
+                WebFetchers.getCustomizableKeyFetchers(
+                                   preferencesService.getImportFormatPreferences(),
+                                   preferencesService.getImporterPreferences())
+                           .stream()
+                           .filter(fetcher -> fetcher.getName().equals(apiKeyName))
+                           .findFirst();
+
+        if (fetcherOpt.isEmpty()) {
+            dialogService.showErrorDialogAndWait(
+                    Localization.lang("Check %0 API Key Setting", apiKeyName),
+                    Localization.lang("Fetcher unknown!"));
+            return;
+        }
+
+        final String testUrlWithoutApiKey = fetcherOpt.get().getTestUrl();
+        if (testUrlWithoutApiKey == null) {
+            dialogService.showWarningDialogAndWait(
+                    Localization.lang("Check %0 API Key Setting", apiKeyName),
+                    Localization.lang("Fetcher cannot be tested!"));
+            return;
+        }
+
+        final String apiKey = selectedApiKeyProperty.get().getKey();
+
+        boolean keyValid;
+        if (!apiKey.isEmpty()) {
+            URLDownload urlDownload;
+            try {
+                SSLSocketFactory defaultSslSocketFactory = HttpsURLConnection.getDefaultSSLSocketFactory();
```

Comment:
Not sure how, but I think this should take the custom SSL store into account

Reply:
I am not completely sure, but I strongly believe that SSL is handled by URLDownload or in the DefaultSSlSocketFactory internally. At least I have no place found in our codebase that handles the custom SSL store directly besides the custom SSL preferences. Also the pr that introduced custom SSL certificates did not introduce any custom SSL handling anywhere, but just removed the infamous bypassSSLVerification.